### PR TITLE
Expand dims op

### PR DESCRIPTION
**Pull request by kahmed10**
This PR also addresses #234 because this may impact expand_dims.

### CODE REVIEW

File: src/tf/tf.cpp [hunk 1]
```diff
@@ -117,17 +118,17 @@ struct tf_parser
...
-            auto new_idx         = parse_axis(i);
+            auto new_idx         = parse_axis(i, new_data.size());
             new_data.at(new_idx) = prev_data.at(i);
         }
         prev_data = new_data;
     }
 
     template <class T>
-    T parse_axis(const T& dim) const
+    T parse_axis(const T& dim, const size_t& num_dims) const
```
Comment (pfultz2) [inline src/tf/tf.cpp]: `num_dims` should be passed by value.

File: src/tf/tf.cpp [hunk 2]
```diff
@@ -165,6 +166,7 @@ struct tf_parser
         add_mem_op("Const", &tf_parser::parse_constant);
         add_mem_op("Conv2D", &tf_parser::parse_conv);
         add_mem_op("DepthwiseConv2dNative", &tf_parser::parse_depthwiseconv);
+        add_mem_op("ExpandDims", &tf_parser::parse_expanddims);
```
Comment (pfultz2) [inline src/tf/tf.cpp]: The transpose flag should be set to false.

File: src/tf/tf.cpp [hunk 3]
```diff
@@ -490,6 +492,25 @@ struct tf_parser
         return prog.add_instruction(op, {l0, new_weights});
     }
 
+    instruction_ref
+    parse_expanddims(const std::string&, const attribute_map&, std::vector<instruction_ref> args)
+    {
+        std::vector<size_t> input_dims = args[0]->get_shape().lens();
+        std::vector<int64_t> new_dims(input_dims.begin(), input_dims.end());
+        size_t num_dims = input_dims.size();
+        int32_t dim     = parse_axis(args[1]->eval().at<int32_t>(), num_dims);
```
Comment (pfultz2) [inline src/tf/tf.cpp]: The `parse_axis` function shouldn't be needed if we set the `transpose` flag to false.

File: src/tf/tf.cpp (hunk 1 above)
Comment (kahmed10) [reply src/tf/tf.cpp]: fixed

File: src/tf/tf.cpp (hunk 2 above)
Comment (kahmed10) [reply src/tf/tf.cpp]: fixed

File: src/tf/tf.cpp (hunk 3 above)
Comment (kahmed10) [reply src/tf/tf.cpp]: fixed

**Pull request merged**